hilti/toolchain/src/compiler/validator.cc

```diff
@@ -354,6 +354,14 @@ struct VisitorPost : visitor::PreOrder, public validator::VisitorMixIn {
 
     void operator()(ctor::Null* n) final {}
 
+    void operator()(ctor::RegExp* n) final {
+        if ( n->attributes()->has(hilti::Attribute::Kind::Anchor) )
+            // This can end up reporting the same location multiple times,
+            // which seems fine. Otherwise we'd need to explicitly track what's
+            // reported already.
```

Comment:
Yeah, this is due to `deprecated` directly emitting output while `error` implicitly deduplicates. The current impl also has the behavior that deprecations are reported before errors which seems backwards. I am not convinced it is worth pulling `deprecation` into `Node` like we do for `error` given the amount of deprecation warnings we currently emit.